pkg/distributor/distributor.go

```diff
@@ -224,7 +225,7 @@ func (d *Distributor) getClientFor(ingester *ring.IngesterDesc) (client.Ingester
 		return client, nil
 	}
 
-	client, err := d.cfg.ingesterClientFactory(ingester.Addr, d.cfg.CompressToIngester)
+	client, err := d.cfg.ingesterClientFactory(ingester.Addr, d.cfg.CompressToIngester, d.cfg.IngesterClientConfig)
```

Comment:
For bonus points you could move the CompressToIngester flags into the IngesterClientConfig.

Reply:
Agreed. I will make the adjustment.